internal/slicer/slicer.go

```diff
@@ -284,27 +214,10 @@ func Run(options *RunOptions) (*Report, error) {
 		return nil
 	}
 	checkRead := func(path string) error {
-		var err error
 		if !knownPaths[path] {
-			// we assume that path is clean and ends with slash if it designates a directory
-			if path[len(path)-1] == '/' {
-				if path == "/" {
-					panic("internal error: content root (\"/\") is not selected")
-				}
-				if knownPaths[path[:len(path)-1]] {
-					err = fmt.Errorf("content is not a directory: %s", path[:len(path)-1])
-				} else {
-					err = fmt.Errorf("cannot list directory which is not selected: %s", path)
-				}
-			} else {
-				if knownPaths[path+"/"] {
-					err = fmt.Errorf("content is not a file: %s", path)
-				} else {
-					err = fmt.Errorf("cannot read file which is not selected: %s", path)
-				}
-			}
+			return checkReadError(knownPaths, path)
```

Comment:
Hmmm.... I see how `checkReadError` is used now. The function alone (down) seemed weird to me at first, not knowing where this will be used.
Maybe it could use more comments, or instead creating a new function, keeping the content inlined as before. The issue with the latter approach is that the `slicer.Run` function will only grow in size.

Maybe running the mutation scripts should be in it's own function?

Reply:
Good suggestion, I will try to make it its own function and see how it looks. My first impressions was that the code was not too lengthy and that we would need to pass a lot of variables (`knownPaths`, `pathInfos`, `slice`, etc.)